wait.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash -xv
+
+function wait-for() {
+  cmd="$1"
+  event="$2"
+  COUNTER=0
+  SLEEP_TIME=${DELAY:-2}
+  MAX_COUNT=${MAX_LOOP:-20}
+  echo "Executing: $cmd"
+
+  while [  $COUNTER -lt 20 ]; do
+    if $cmd; then
+      echo "$event succeeded"
+      break;
+    else
+      COUNTER=$(($COUNTER + 1))
+      echo "Waiting $SLEEP_TIME sec for $event to succeed another $(($MAX_COUNT - $COUNTER)) times"
+      sleep $SLEEP_TIME
+    fi
+  done
+  if [ $COUNTER -eq $MAX_COUNT ]; then
+    echo "Giving up on $event ..........."
+    return 1
+  fi
+}
+
+function is-solr-running() {
+  wget -q -O - "$solr_url" | grep -q -i solr
+}
+
+function is-solr-cluster-number_reached() {
+  [ $(curl "$solr_url"/solr/admin/collections?action=clusterstatus | jq '.cluster.live_nodes[]' | wc -l) -gt 2 ]
+}
+
+function is-solr-collection-creation-status-completed() {
+  curl "$solr_url/solr/admin/collections?action=REQUESTSTATUS&requestid=5555&wt=json" | jq '.status.state' | grep -q 'completed'
+}
+
+function create-config() {
+  config_name="$1"
+  if curl "$solr_url/solr/admin/configs?action=CREATE&name=$config_name&baseConfigSet=mongoConnectorBaseConfig&configSetProp.immutable=false&wt=json&omitHeader=true" | jq ".error != null" | grep -q true; then
+    echo "Failed creating $config_name, Exiting ..."
+    exit 1
+  fi
```

Comment:
do you really want to exit 1 and not return 1?

Reply:
yes this way k8s will relaunch the pod up to back off limit times